runtime/pack/dist/opt/termdebug/plugin/termdebug.vim

```diff
@@ -630,76 +630,86 @@ function! s:OpenHoverPreview(lines, filetype) abort
+      let width = 0
+      for index in range(len(lines))
+        let line = lines[index]
+        if line !=# ''
+          " Give a left margin
+          for i in range(margin)
+            let line = ' ' . line
+          endfor
         endif
-
-        " Prefer West, but if there is no space, fallback into East
-        if pos[2] + width <= &columns
-            let hor = 'W'
-            let col = 0
-        else
-            let hor = 'E'
-            let col = 1
+        let lw = strdisplaywidth(line)
+        if lw > width
+          let width = lw
         endif
+        let lines[index] = line
+      endfor
 
-        let float_win_id = nvim_open_win(bufnr, v:true, {
-        \   'relative': 'cursor',
-        \   'anchor': vert . hor,
-        \   'row': row,
-        \   'col': col,
-        \   'width': width,
-        \   'height': height,
-        \ })
+      " Give margin
+      let width += margin
+      for i in range(margin)
+        let lines = [''] + lines + ['']
+      endfor
+      let height = len(lines)
+
+      " Calculate anchor
+      " Prefer North, but if there is no space, fallback into South
+      let bottom_line = line('w0') + winheight(0) - 1
+      if pos[1] + height <= bottom_line
+        let vert = 'N'
+        let row = 1
+      else
+        let vert = 'S'
+        let row = 0
+      endif
+
+      " Prefer West, but if there is no space, fallback into East
+      if pos[2] + width <= &columns
+        let hor = 'W'
+        let col = 0
+      else
+        let hor = 'E'
+        let col = 1
+      endif
 
-        execute 'noswapfile edit!' bufname
+      let buf = nvim_create_buf(v:false, v:true)
+      call nvim_buf_set_lines(buf, 0, -1, v:true, lines)
+      " using v:true for second argument of nvim_open_win make the floating
+      " window disappear
+      let float_win_id = nvim_open_win(buf, v:false, {
+            \   'relative': 'cursor',
+            \   'anchor': vert . hor,
+            \   'row': row,
+            \   'col': col,
+            \   'width': width,
+            \   'height': height,
+            \ })
+      call nvim_win_set_option(float_win_id, 'relativenumber', v:false)
+      call nvim_win_set_option(float_win_id, 'signcolumn', 'no')
+      call nvim_win_set_option(float_win_id, 'signcolumn', 'no')
+      if a:filetype isnot v:null
+        call nvim_win_set_option(float_win_id, 'filetype', a:filetype)
+      endif
 
-        setlocal winhl=Normal:CursorLine
+      " cannot use nvim_win_set_option for these options
+      setlocal nomodified nomodifiable
+
+      " Unlike preview window, :pclose does not close window. Instead, close
+      " hover window automatically when cursor is moved.
+      let call_after_move = printf('<SID>CloseFloatingHoverOnCursorMove(%d, %s)', float_win_id, string(pos))
+      let call_on_bufenter = printf('<SID>CloseFloatingHoverOnBufEnter(%d, %d)', float_win_id, bufnr)
+      augroup termdebug-close-hover
```

Comment:
let's name this `nvim_termdebug`. We already define the `nvim_terminal` augroup, so let's stay consistent with the `nvim_` prefix.